Documentation/SwiftlyDocs.docc/swift-version.md

```diff
@@ -0,0 +1,97 @@
+
+Luckily, swiftly exists to make it easy to get a toolchain. Which one of the potentially large number do I pick? This question can be bewildering.
+
+## What is the workflow?
+
+Your workflow can be one way to decide what toolchain you should use.
+
+### Getting started working on a project
+
+When you're working on a particular project, such as developing a bug fix, or adding a feature, you'll want the toolchain that sets you up on a path to successfully getting your pull request merged at that point in time. Along the way, you are likely to need a toolchain that will successfully compie the code with the language features that it requires, and is also unlikely to have show-stopping problems in doing that.
+
+Iterating on this code might involve using a supported IDE, such as VS Code, that makes use of a matching language server (sourcekit-lsp) that works with this code. Perhaps the code base uses certain features from Swift Testing, only available starting in a particular release. The version of swift-format can also have an impact on the formatting (and the formatting checks) on the code, so it will save time and increase confidence when formatting your new code to use the correct version. A developer would like the version that's most likely to succeed in all of these areas.
+
+A project can publish the recommended default version of the specific toolchain in their developer's guide. This is a good start, and helps someone to get started with a good chance of success. Documention is often a worthwhile investment, but sometimes we can do better with tooling.
+
+### Verifying a project
+
+When verifying a project in something like a CI system, it's important to pick a toolchain that meets the objective of verifying that the current state of the code with a level of reliability. Using the [test pyramid](https://martinfowler.com/articles/practical-test-pyramid.html) model, you want to get the *majority* of tests running _quickly_, but also *reliably*.
+
+Reliability in toolchain selection is knowing which toolchain has produced good results in the pastat a developer's desk, better would be to use one that's been verified in CI. And so, maybe the CI system uses the toolchain that's documented in the developer's guide? It can be a manual process, such as when the developer's guide is updated. Better would be to have some tooling support that handles both develpers getting started and toolchain selection in CI. In any case swiftly can help install the chosen default toolchain.
+
+It is important to note it is often important to engage in certain kinds of testing that can be slower, and less reliable. The test pyramid model accommodates ths for doing integration, end-to-end testing. These should be fewer in number, and possibly less frequently. Since these tests can be less reliable and slower, sometimes they don't run with every pull request. They are commonly run on a schedule with processes in place to react to failures.
```

Comment:
This bit feels off topic from the goal of the document (what toolchain should I use)